fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/RecordCursorTest.java

```diff
@@ -1158,8 +1158,16 @@ public void reduceTest() throws Exception {
         cursor = RecordCursor.fromList(Arrays.asList(1, 2, 3, 4, 5, 6, 7));
         sum = cursor.reduce(0, Integer::sum, x -> x > 10);
         assertEquals(15, sum.get());
+    }
 
-
+    @Test
+    public void asStreamTest() throws Exception {
+        assertEquals(7, RecordCursor.fromList(Arrays.asList(1, 2, 3, 4, 5, 6, 7)).asStream().count());
+        assertEquals(2, RecordCursor.fromList(Arrays.asList(2, 3)).asStream().findFirst().get());
+        assertThrows(RuntimeException.class, () -> RecordCursor.fromList(Arrays.asList(2, 3)).asStream(
+                        () -> {
+                            throw new RuntimeException("on close");
+                        }).close());
```

Comment:
Could you also add a test to verify that a ```MapCursor``` got by calling ```RecordCursor.map(@Nonnull Function<T, V> func)``` can also generate a ```stream``` that outputs expected mapped elements?

Reply:
Done.